### PR TITLE
sets read_write_scope from opts, this will permit to comply to readOn…

### PR DESCRIPTION
**Pull request by juantascon**
…ly parameters

### CODE REVIEW

**Comment by riccardomanfrin** [comment]
Thumbup, makes sense to me

**Review by albertored — CHANGES_REQUESTED** [review]
Can you add the same option to the `OpenApiSpex.Cast.cast/4` function and to the the `OpenApiSpex.cast_value/3,4` ones? 

Also I just noticed that the `:apply_defaults` option that is available in `OpenApiSpex.Cast.cast/4` is not available in `OpenApiSpex.cast_value/3,4`, can you add it in the context of this PR, it should be straightforward.

**Comment by juantascon** [comment]
I don't get this, OpenApiSpex.Cast.cast/4 already has the parameter, also there is no OpenApiSpex.cast_value/4 and OpenApiSpex.cast_value/3 doesn't have a opts parameter

File: lib/open_api_spex/cast.ex [hunk 1]
```diff
@@ -103,7 +103,16 @@ defmodule OpenApiSpex.Cast do
   @spec cast(schema_or_reference | nil, term(), map(), [cast_opt()]) ::
```
Comment (albertored) [inline lib/open_api_spex/cast.ex:103]: `cast_opt()` type should include the new option

**Pull request merged**